File: Archive/AtCoder/Boot_camp_for_Beginners/Easy/B - Qualification simulator.py
n,a,b=map(int,input().split())
s=input()
pa=pb=0
for si in s:
    if si=="a" and pa+pb<a+b:
        pa+=1
        print("Yes")
    elif si=="b" and pa+pb<a+b and pb<b:
        pb+=1
        print("Yes")
    else:
        print("No")

File: Archive/AtCoder/Boot_camp_for_Beginners/Easy/B - Qualification simulator.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
  int N,A,B; cin>>N>>A>>B;
  string S; cin>>S;
  int p=0,o=0;
  for(char Si:S){bool f=0;
    if(Si=='a')if(p<A+B)f=1;
    if(Si=='b'){o++;if(p<A+B&&o<=B)f=1;}
    if(f){cout<<"Yes"<<endl; p++;} else cout<<"No"<<endl;
  }
}